src/scheduler/task.rs

```diff
@@ -559,6 +565,15 @@ impl BlockedTaskQueue {
 				first_task = false;
 				cursor.move_next();
 			}
+
+			// No, then just insert it at the end of the list.
+			self.list.push_back(new_node);
+
+			// If this is the new first task in the list, update the One-Shot Timer
+			// to fire when this task shall be woken up.
+			if first_task {
+				arch::set_oneshot_timer(wakeup_time);
+			}
```

Comment:
Can you use scopeguards for code like this? https://docs.rs/scopeguard/1.1.0/scopeguard/ This would also get rid of the duplicated code and prevent the same issue from reappearing after a change to the function. I did not look into it in detail, not sure if it is really a good solution.

Reply:
Good idea, I integrate a scopeguard